packages/flutter_tools/test/src/package_config.dart

```diff
@@ -0,0 +1,49 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+import 'package:file/file.dart';
+import 'package:flutter_tools/src/globals.dart' as globals;
+
+/// Writes a `.dart_tool/package_config.json` file at [directory].
+///
+/// If directory is not specified, it will be `globals.fs.currentDirectory`;
+///
+/// It will contain a package entry for [mainLibName] with `rootUri` at
+/// [directory].
+///
+/// [otherLibs] maps other package names to their `rootUri` relative to `directory`.
+///
+/// Returns the `File` Object representing the package config.
+File writePackageConfigFile({
```

Comment:
This is more of a clarification question than a block:

Why do this instead of just `Iterable<Package> packages` (from `package:package_config`)?

Is there something that is gained other than a bit of code tersity?

i.e. https://github.com/flutter/flutter/blob/8ad78cce13cf39e71398c9016d84f42b6bc091ae/packages/flutter_tools/test/src/package_config.dart.

Reply:
I did not really consider using a list of `Package` but looking at the constructor, it requires repeating the `root` uri (the location of the package config)
I think that makes it kinda unergonomic.

Also the `Package.packageUri` getter is always absolute, meaning no easy way to test relative paths.

I think I'll keep this for now, though I can see how a list of packages could make sense